library/cohesity_generic_nas_job.py

```diff
@@ -0,0 +1,427 @@
+        raise__cohesity_exception__handler
+
+
+cohesity_client = None
+SLEEP_TIME_SECONDS = 60
+
+
+def get_generic_nas_job_details(module):
+    '''
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        jobs = cohesity_client.protection_jobs.\
+            get_protection_jobs(names=module.params.get('job_name'))
+        for job in jobs:
+            if job.name == module.params.get('job_name'):
+                return job
+        return None
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_policy_id(module):
+    '''
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        policies = cohesity_client.protection_policies.\
+            get_protection_policies(names=module.params.get('protection_policy'))
+        for policy in policies:
+            if policy.name == module.params.get('protection_policy'):
+                return policy.id
+        raise__cohesity_exception__handler(
+            "Failed to find protection policy " + module.params.get('protection_policy'), module)
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_view_box_id(module):
+    '''
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        view_boxes = cohesity_client.view_boxes.get_view_boxes(names=module.params.get('storage_domain'))
+        for view_box in view_boxes:
+            if view_box.name == module.params.get('storage_domain'):
+                return view_box.id
+        raise__cohesity_exception__handler(
+            "Failed to find storage domain " + module.params.get('storage_domain'), module)
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_generic_nas_parent_id(module):
+    '''
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        sources = cohesity_client.protection_sources.list_protection_sources_root_nodes(environments='kGenericNas')
+        for source in sources:
+            return source.protection_source.id
+    except APIException as ex:
+        raise__cohesity_exception__handler(str(
+            json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_generic_nas_source_ids(module):
```

Comment:
not necessary, but any helper functions start it with _ , this is to just distinguish the functions

Reply:
ok